ash-window/src/lib.rs

```diff
@@ -171,3 +171,102 @@ pub fn enumerate_required_extensions(
+            // https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap34.html#platformQuerySupport_walyand
+            let ext = khr::WaylandSurface::new(entry, instance);
+            Ok(ext.get_physical_device_wayland_presentation_support(
+                physical_device,
+                queue_family_index,
+                h.display,
+            ))
+        },
+        RawDisplayHandle::Windows(_) => unsafe {
+            // https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap34.html#platformQuerySupport_win32
+            let ext = khr::Win32Surface::new(entry, instance);
+            Ok(ext.get_physical_device_win32_presentation_support(
+                physical_device,
+                queue_family_index,
+            ))
+        },
+        #[cfg(unix)]
+        RawDisplayHandle::Xcb(h) => unsafe {
+            // https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap34.html#platformQuerySupport_xcb
+            let ext = khr::XcbSurface::new(entry, instance);
+
+            let xcb = xcb::Connection::from_raw_conn(h.connection.cast());
+            let setup = xcb.get_setup();
+            let screen = setup.roots().nth(h.screen as usize).unwrap();
+            let visual = screen.root_visual();
+            xcb.into_raw_conn();
+
+            let res = ext.get_physical_device_xcb_presentation_support(
+                physical_device,
+                queue_family_index,
+                h.connection,
+                visual,
+            );
+
+            Ok(res)
+        },
+        #[cfg(unix)]
+        RawDisplayHandle::Xlib(h) => unsafe {
+            // https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap34.html#platformQuerySupport_xlib
+            let ext = khr::XlibSurface::new(entry, instance);
+
+            #[cfg(not(any(feature = "x11-dl", feature = "x11")))]
+            compile_error!("Exactly one of x11-dl or x11 must be enabled on unix");
+            #[cfg(all(feature = "x11-dl", feature = "x11"))]
+            compile_error!("Exactly one of x11-dl or x11 must be enabled on unix");
+
+            let visual_id;
+            #[cfg(feature = "x11-dl")]
+            {
+                let xlib = x11_dl::xlib::Xlib::open().unwrap();
+                let default_visual = (xlib.XDefaultVisual)(h.display.cast(), h.screen);
+                visual_id = (xlib.XVisualIDFromVisual)(default_visual);
+            }
+            #[cfg(feature = "x11")]
+            {
+                let default_visual = x11::xlib::XDefaultVisual(h.display.cast(), h.screen);
+                visual_id = x11::xlib::XVisualIDFromVisual(default_visual);
+            }
+
+            Ok(ext.get_physical_device_xlib_presentation_support(
+                physical_device,
+                queue_family_index,
+                h.display,
+                visual_id as _,
+            ))
+        },
+        // All other platforms mentioned in the vulkan spec don't
```

Comment:
https://github.com/ash-rs/ash/pull/774#discussion_r1377297481

```suggestion
        // All other platforms mentioned in the Vulkan spec don't
```